server/services/mapping_service.py

```diff
@@ -96,6 +89,29 @@ def unlock_task_after_mapping(mapped_task: MappedTaskDTO) -> TaskDTO:
         return task.as_dto()
 
     @staticmethod
+    def stop_mapping_task(stop_task: StopMappingTaskDTO) -> TaskDTO:
+        """ Unlocks the task and sets the task history appropriately """
+        task = MappingService.get_task_to_unlock(stop_task.project_id, stop_task.task_id, stop_task.user_id)
+
+        if stop_task.comment:
+            # Parses comment to see if any users have been @'d
+            MessageService.send_message_after_comment(stop_task.user_id, stop_task.comment, task.id,
+                                                      stop_task.project_id)
+
+        task.reset_lock(stop_task.user_id, stop_task.comment)
+        return task.as_dto()
+
+    @staticmethod
+    def get_task_to_unlock(project_id, task_id, user_id):
+        task = MappingService.get_task(task_id, project_id)
```

Comment:
Nice
Suggest renaming to get_task_locked_by_user
Add a docstring here describing function and good to go